sdk/src/poh_config.rs

```diff
@@ -15,10 +15,30 @@ pub struct PohConfig {
     /// None enables "Low power mode", which implies:
     /// * sleep for `target_tick_duration` instead of hashing
     /// * the number of hashes per tick will be variable
-    pub hashes_per_tick: Option<u64>,
+    hashes_per_tick: Option<u64>,
```

Comment:
Probably not great to make one field of a simple structure different from others. Maybe make all fields private with getter/setter? But that'd probably touch a lot of code. 

On the other hand, if this PR's goal is to make it possible to default `poh_config.hashes_per_tick` to different values based on a feature gate, would it be possible to leave `poh_config` unchanged, but introduce a function that does something like this
```
fn get_default_hashes_per_tick( feature_set ) -> u64 {
    if feature_set.is_enabled(...) {
        NEW_DEFAULT_HASHES_PER_TICK
    } else {
        clock::DEFAULT_HASHES_PER_TICK
    }
} 
```
Then use `get_default_hashes_per_tick()` wherever constant is used today.

Reply:
Yeah, interesting idea for sure, but it would be very easy for a future developer to just set any value they want at any point, and if they didn't use get_default_hashes_per_tick, things wouldn't work. The private member is the only way I could see to make this future-programmer-proof.
As for also making the other values private.. yeah, I agree it's a little unbalanced-looking to have one priv and the others pub, but there's nothing really wrong with that, and I didn't want to touch another set of files for a change that didn't do anything we need.

Reply:
@KirillLykov , I'll make the naming change unless we (@taozhu-chicago) decide that there's a way to make get_default() to work sensibly. Because if so, that would be a much smaller code change.

Reply:
Yeah, this is more about style and so on, so the best approach to do similar to the other parts of the code, I guess.